src/Task/CommandStack.php

```diff
@@ -105,12 +105,16 @@ public function run()
             throw new TaskException($this, 'You must add at least one command');
         }
         if (!$this->stopOnFail) {
-            $this->printTaskInfo('{command}', ['command' => $this->getCommand()]);
+            if ($this->isMetadataPrinted) {
```

Comment:
Should probably add a method to the trait rather than accessing the property directly.